src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
@lambday 
Yes. we can use `CKernel::get_kernel_diagonal()`
However, I may do some similar tricks when the diagonal elements of derivatives wrt hyper-parameter are required. say `get_parameter_gradient_diagonal(param, i)` if needed.

see `deriv_trtr=m_kernel->get_parameter_gradient(param, i).get_diagonal_vector();`